rtengine/iimage.h

```diff
@@ -415,6 +415,21 @@ class PlanarWhateverData : virtual public ImageDatas
                     v(y, x) = tmp;
                 }
             }
+
+            // Middle row of odd-height images: only go half way otherwise the
+            // pixels will be swapped twice.
+            if (height & 1) {
+                int i = height / 2;
+                int width2 = width / 2;
+                for (int j = 0; j < width2; j++) {
+                    T tmp;
+                    int x = width - 1 - j;
+
+                    tmp = v(i, j);
```

Comment:
@Lawrence37 Isn't that a prime example for [`std::swap()`](https://en.cppreference.com/w/cpp/algorithm/swap)? And if not, I'd rather have it `const T tmp = v(i, j);`...

Reply:
I was being lazy (and cautious) and copy-pasted the code from above. I will see if swap is usable here.

Reply:
Replaced with `std::swap`